### PR TITLE
Epic 6798 : List Domain Designer Conversion

### PR DESCRIPTION
**Pull request by labkey-bpatel**
Updates related to allowing binding of DomainKind specific options/properties to a bean & generifying DomainKind.

### CODE REVIEW

**Review by labkey-jeckels — APPROVED** [review]

File: api-src/org/labkey/api/snd/SNDDomainKind.java [hunk 1]
```diff
@@ -24,6 +24,7 @@
 import org.labkey.data.xml.domainTemplate.DomainTemplateType;
 import org.labkey.data.xml.domainTemplate.SNDTemplateType;
 
+import org.json.JSONObject;
```
Comment (labkey-jeckels) [inline api-src/org/labkey/api/snd/SNDDomainKind.java]: Unused import

File: api-src/org/labkey/api/snd/PackageDomainKind.java [hunk 2]
```diff
@@ -25,6 +25,8 @@
 import org.labkey.api.security.User;
 import org.labkey.api.security.permissions.AdminPermission;
 
+import org.json.JSONObject;
```
Comment (labkey-jeckels) [inline api-src/org/labkey/api/snd/PackageDomainKind.java]: Unused import

**Review by labkey-jeckels — APPROVED** [review]
OK to merge after cleaning up unused imports

File: api-src/org/labkey/api/snd/PackageDomainKind.java (hunk 2 above)
Comment (labkey-bpatel) [reply api-src/org/labkey/api/snd/PackageDomainKind.java]: done!

File: api-src/org/labkey/api/snd/SNDDomainKind.java (hunk 1 above)
Comment (labkey-bpatel) [reply api-src/org/labkey/api/snd/SNDDomainKind.java]: done!

**Pull request merged**